app/controllers/cases/offender_sar_controller.rb

```diff
@@ -211,7 +211,11 @@ def has_optional_flags?
     end
   
     def back_link_url
-      "#{@case.case_route_path}/#{@case.get_previous_step}#{build_url_params_from_flags}"
+      if @case.get_previous_step.nil?        
+        new_case_path
+      else
+        "#{@case.case_route_path}/#{@case.get_previous_step}#{build_url_params_from_flags}"
+      end
```

Comment:
This is a tiny style thing, but it's weird to ask if something is nil in Ruby - we prefer to phrase it the other way round so we're asking if a thing exists. How would you feel about something like this?
 
```
        if @case.get_previous_step        
          "#{@case.case_route_path}/#{@case.get_previous_step}#{build_url_params_from_flags}"
        else
          new_case_path
        end
```